### PR TITLE
Deprecate libmesh_make_unique

### PR DESCRIPTION
**Pull request by roystgnr**
We can just use `std::make_unique` now.

I also got rid of `#include "libmesh/auto_ptr.h"` in headers that only needed `<memory>` - though that might break any downstream code that's still using `libmesh_make_unique` and relying on indirect header inclusion to find it.

### CODE REVIEW

File: include/base/auto_ptr.h
Comment (jwpeterson) [inline include/base/auto_ptr.h:35]: I would vote to remove the `UniquePtr` alias now as well. Every once in a while we get copy/pasted code from ancient examples that incorrectly re-introduces `UniquePtr` usage, so it would be good to completely avoid that. I have a branch which does this, but it will likely conflict with this larger branch, so maybe better to do as part of this PR?

**Review by jwpeterson — APPROVED** [review]
This looks good (and long overdue) to me. Are you waiting on any other CI testing before merging?

File: include/base/auto_ptr.h
Comment (jwpeterson) [reply include/base/auto_ptr.h:35]: jwpeterson/libmesh@9834ba282458c60bd96796d9f233d78f97c40916

Comment (roystgnr) [reply include/base/auto_ptr.h:35]: I agree; we've had it deprecated forever now.  I'd prefer a separate PR though.  I'll do that if I've inadvertently trampled on yours.

**Comment by roystgnr** [comment]
Nah; I was waiting on the Moose Mac CI to be fixed, and now that that's done and passed we're good to go.

**Pull request merged**
